### PR TITLE
Create polynomial_hash.py

### PR DESCRIPTION
**Pull request by prasad-chavan1**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: hashes/polynomial_hash.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    p_pow = 1
+    for char in s:
+        char_value = ord(char) - ord("a") + 1  # Convert character to a numerical value
+        hash_value = (hash_value + char_value * p_pow) % m
+        p_pow = (p_pow * p) % m
+    return hash_value
+
+
+print(polynomial_hash("PythonLanguage"))
+# Output: 877483825
```
Comment (tianyizheng02) [inline hashes/polynomial_hash.py:25]: Please encapsulate all driver code in a `__main__` block.

File: hashes/polynomial_hash.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"""
+Calculate the hash of a string using a polynomial rolling hash function.
```
Comment (tianyizheng02) [inline hashes/polynomial_hash.py:2]: If this is a polynomial rolling hash, then why not name the file and function `polynomial_rolling_hash`?

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
What is the hashing algorithm here?

**Comment by cclauss** [comment]
Closing require_type_hints PRs to prepare for Hacktoberfest 2024

**Pull request closed without merging**